File: problems/Medium/78. Subsets.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        int n=pow(2,nums.size());
        vector<vector<int>> ans;
        for(int i=0;i<n;++i){
            vector<int>temp;
            int x=i,c=0;
            while(x){
                if(x&1) temp.push_back(nums[c]);
                c++;
                x/=2;
            }
            ans.push_back(temp);
        }
        return ans;
    }
};

File: problems/Medium/78. Subsets.go
func subsets(nums []int) [][]int {
    len := float64(len(nums))
    n := int(math.Pow(2, len))
    var res [][]int
    for i:=0 ; i<n ; i++ {
        var tmp []int
        ind := 0
        val := i
        for val > 0 {
            if val & 1 > 0 {
                tmp = append(tmp, nums[ind])
            }
            val = val/2
            ind = ind + 1
        }
        res = append(res, tmp)
    }
    return res
}